synapse/events/__init__.py

```diff
@@ -257,7 +259,16 @@ def auth_event_ids(self):
         return [e for e, _ in self.auth_events]
 
 
-class FrozenEvent(EventBase):
+class FrozenEventBase(EventBase):
```

Comment:
actually, I'm a bit unconvinced by this, not least because its name is a bit poor. What makes it different to EventBase?

Reply:
`EventBase` won't have an `event_id` if its a builder. I guess we could shove it into `EventBase` and just have the builder explode, but that sounds a little bit silly. Really, I think we should be better at differentiating with types what we expect to be actual events, vs event builders

Reply:
a builder? as in `EventBuilder` ? that's not an `EventBase` ?

Reply:
Wait, it isn't? It isn't! Used to be though. Ok this can now die in a fire :fire:

Reply:
:tada: :fire: :boom: